src/Analyzers/Core/CodeFixes/RemoveUnnecessarySuppressions/RemoveUnnecessaryAttributeSuppressionsCodeFixProvider.cs

```diff
@@ -30,27 +33,26 @@ public override ImmutableArray<string> FixableDiagnosticIds
         internal override CodeFixCategory CodeFixCategory
             => CodeFixCategory.CodeQuality;
 
-        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
+            var root = await context.Document.GetRequiredSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
             foreach (var diagnostic in context.Diagnostics)
             {
-                context.RegisterCodeFix(
-                    new MyCodeAction(c => FixAsync(context.Document, diagnostic, c)),
-                    diagnostic);
+                if (root.FindNode(diagnostic.Location.SourceSpan) != null)
```

Comment:
why was this needed?

Reply:
Defensive check if some other analyzer reports a diagnostic with same ID.

Reply:
doc.